tests/unit/lms/services/canvas_api/client_test.py

```diff
@@ -398,6 +434,10 @@ def test_public_url(self, canvas_api_client, http_session):
             timeout=Any(),
         )
 
+    @pytest.fixture
+    def registry(self):
+        return create_autospec(Registry, instance=True, spec_set=True)
```

Comment:
Another way to go here is to use:

```python
with patch.object(pyramid_request, "registry") as registry:
    yield registry
```

All the same really, but that has the advantage of not having to work out what the class is. This is very clear though (as long as the class is right!).